storage/src/multiple_strategy_block_store.rs

```diff
@@ -0,0 +1,109 @@
+// A mixed block store,
+// Stores confirmed blocks in memory
+// Finalized blocks in long term storage of your choice
+// Fetches legacy blocks from faithful
+
+use crate::block_stores::inmemory_block_store::InmemoryBlockStore;
+use async_trait::async_trait;
+use solana_lite_rpc_core::block_storage_interface::{BlockStorageInterface, BlockStorageImpl};
+use solana_rpc_client::nonblocking::rpc_client::RpcClient;
+use solana_rpc_client_api::config::RpcBlockConfig;
+use solana_sdk::{commitment_config::{CommitmentConfig, CommitmentLevel}, slot_history::Slot};
+use solana_transaction_status::{TransactionDetails, UiTransactionEncoding, UiConfirmedBlock};
+use std::sync::{Arc, atomic::{AtomicU64, Ordering}};
+
+pub struct MultipleStrategyBlockStorage {
+    inmemory_for_storage: InmemoryBlockStore, // for confirmed blocks
+    persistent_block_storage: BlockStorageImpl, // for persistent block storage
+    faithful_rpc_client: Arc<RpcClient>,      // to fetch legacy blocks from faithful
+    number_of_slots_in_memory: usize,
+    last_confirmed_slot: Arc<AtomicU64>,
+}
+
+impl MultipleStrategyBlockStorage {
+    pub fn new(
+        persistent_block_storage: BlockStorageImpl,
+        faithful_url: String,
+        number_of_slots_in_memory: usize,
+    ) -> Self {
+        Self {
+            inmemory_for_storage: InmemoryBlockStore::new(number_of_slots_in_memory),
+            persistent_block_storage,
+            faithful_rpc_client: Arc::new(RpcClient::new(faithful_url)),
+            number_of_slots_in_memory,
+            last_confirmed_slot: Arc::new(AtomicU64::new(0)),
+        }
+    }
+
+    pub async fn get_in_memory_block(&self, slot: Slot) -> Option<UiConfirmedBlock> {
+        self.inmemory_for_storage.get(slot).await
+    }
+}
+
+#[async_trait]
+impl BlockStorageInterface for MultipleStrategyBlockStorage {
+    async fn save(
+        &self,
+        slot: solana_sdk::slot_history::Slot,
+        block: solana_transaction_status::UiConfirmedBlock,
+        commitment: solana_sdk::commitment_config::CommitmentLevel,
+    ) {
+        if commitment == CommitmentLevel::Confirmed {
+            self.inmemory_for_storage
+                .save(slot, block, commitment)
+                .await;
+        } else if commitment == CommitmentLevel::Finalized {
+            let block_in_mem = self.inmemory_for_storage.get(slot).await;
+            if block_in_mem.is_none() {
```

Comment:
You should add the contains method to the Inmemory storage and use it to define if the block exists.
Otherwise, you can write it:
```
if let None = self.inmemory_for_storage.get(slot).await {...}
```

Reply:
Ok, it wasn't in my version. I think you should use match instead of if none else if let some. It's easier to read.